upup/pkg/fi/cloudup/azuretasks/loadbalancer.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuretasks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-06-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"k8s.io/klog/v2"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/azure"
+)
+
+//go:generate fitask -type=LoadBalancer
+
+// LoadBalancer is an Azure Cloud LoadBalancer
+type LoadBalancer struct {
+	Name          *string
+	Lifecycle     *fi.Lifecycle
+	ResourceGroup *ResourceGroup
+	Subnet        *Subnet
+
+	// RequirePublicIP is set to true when the Loadbalancer requires a public IP.
+	RequirePublicIP *bool
+	// PublicIPName is the name of the ip created to be attached to the loadbalancer
+	PublicIPName *string
```

Comment:
likely related to my first comment about whether we support `LoadBalancerTypePublic` but I don't see this field set by azuremodel nor do I see it being set by Find below. Both will need to set the field.

Reply:
I can remove this with the rest of the public ip logic. If we keep the logic we'll run into the same issue we have with LoadBalancerName in the VMSS so this may just be better to remove all together.